share/eds/p2p/server.go

```diff
@@ -0,0 +1,154 @@
+package p2p
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"time"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/network"
+	"github.com/libp2p/go-libp2p-core/protocol"
+
+	"github.com/celestiaorg/celestia-node/share/eds"
+	p2p_pb "github.com/celestiaorg/celestia-node/share/eds/p2p/pb"
+	"github.com/celestiaorg/go-libp2p-messenger/serde"
+)
+
+const (
+	// writeDeadline sets timeout for sending messages to the stream
+	writeDeadline = time.Second * 5
+	// readDeadline sets timeout for reading messages from the stream
+	readDeadline = time.Minute
```

Comment:
should make them configurable.

Reply:
In this PR. The reasoning is the same for including protocolSuffix earlier.

Reply:
It's not a bloat. It's a tech debt that we should cover sooner rather than later. 

If we ever stick to some pattern, every new component should follow it. If we introduce a new feature and we know how we want to configure it, we should introduce it with configuration.

Originally, we didn't do it and have to spend some time to catch up everywhere. We really don't want to pile up more things.

I am fine though to extract this into separate PR, which should be part of the epic and come right after.

Reply:
Let's open a discussion about making read and write deadlines configurable. There is absolutely no reason for it, and if we only want to do it for consistency then we need to discuss if it even makes sense in general. Won't happen in this PR, but looking forward to figuring out what's best for the whole codebase, and going from there

Reply:
It's not just about read/write and also about protocolPrefix. We might also have rate-limiting further with configurations, which can simply add a new config param instead of implementing the whole thing. The question is not about read or write, but "does server need configuration or not?" If so, the feature should include it

Reply:
`protocolPrefix` is a very different granularity than read/write deadlines...

Reply:
Explain

Reply:
@renaynay, explain

Reply:
Let's add the complexity that comes with more fine-grained configuration first once we have things that make sense to configure. The configurability of the protocolID matches the rest of our codebase right now.

Reply:
@Wondertan  - protcolPrefix configures who the node can talk to / who talks to the node on that specific protocol (not granular, it is about distinguishing this specific protocol from others) whereas read/write deadlines is a very granular configuration option on the stream level.

We should allow for customisability to the extent that it matters (protocolIDs) but (until we have a VERY good reason for it) not for params as granular as reading/writing to the stream. The defaults are good enough and if someone really wants to toy with the granularity, they can change it in the code. I am against bloat as much as possible unless there's a properly compelling reason for it and "does server need configuration or not?" argument is too generic bc configuration is a spectrum. 

Anyway no need to waste more time discussing this, I hope we can all agree on a general philosophy about this and not have to discuss it again.

Reply:
We should discuss this to the point when all of us agree. At this point, I don't see a configuration being a bloat. Configuration is something that each new core feature should have when it's implemented. However, I don't want to block this PR anymore and force anyone. 

Another performance-related configuration that we _have to_ expose eventually: https://github.com/celestiaorg/celestia-node/pull/1431#discussion_r1053143556

Reply:
> We should discuss this to the point when all of us agree. 

My argument is that configurability exists on a spectrum and movement towards extreme customisation should be justified as it adds additional LOCs. I see justification as 1. several users ask for this, or 2. the param could be a **critical** testing variable.

> At this point, I don't see a configuration being a bloat. 

I disagree -- each additional customisable parameter introduces code bloat ( **objectively** ) in LOCs.

> Configuration is something that each new core feature should have when it's implemented

Also disagree - a feature could be implemented with defaults and tested for correctness, and *later* should come optimisations via greater configurability. The idea that every feature should be implemented from the start with configurability slows down dev iteration process and is just unnecessary.

Reply:
KISS, YAGNI is all I have to say here. Implement it when its needed.

Reply:
implemented params